test/adapter/z-stack/adapter.test.ts

```diff
@@ -3381,13 +3431,69 @@ describe("zstack-adapter", () => {
             [{attrId: 0, attrData: 2, dataType: 32, status: 0}],
             {},
         );
+        const responseBuffer = responseFrame.toBuffer();
         const object = mockZpiObject(Type.AREQ, Subsystem.AF, "incomingMsgExt", {
+            groupid: 12,
             clusterid: 0,
+            srcaddrmode: 2,
+            srcaddr: "0x1234567812345678",
             srcendpoint: 20,
-            srcaddr: 2,
+            //srcpanid
+            dstendpoint: 1,
+            //wasbroadcast
             linkquality: 101,
+            //securityuse
+            timestamp: 0x12345678,
+            //transseqnumber
+            len: responseBuffer.length,
+            data: responseBuffer,
+        });
+        adapter.on("zclPayload", (p) => {
+            zclData = p;
+        });
+
+        znpReceived(object);
+
+        expect(zclData.endpoint).toStrictEqual(20);
+        expect(zclData.groupID).toStrictEqual(12);
+        expect(zclData.linkquality).toStrictEqual(101);
+        expect(zclData.address).toStrictEqual(0x5678);
+        expect(zclData.groupID).toStrictEqual(12);
+        expect(zclData.data).toStrictEqual(Buffer.from([24, 100, 1, 0, 0, 0, 32, 2]));
+        expect(zclData.header.commandIdentifier).toBe(1);
+    });
+
+    it("Incoming message extended (inter PAN) - 64-bit address", async () => {
```

Comment:
Since code is almost the same, could you merge this with the 16-bit test case? This should save a lot of code duplication. The same can be done for the huge bytes test cases.

Example on how to do this: https://github.com/Koenkk/zigbee-herdsman/blob/fe9651586c3ad27869e6e8ee735a40fc1fc0cd9a/test/zcl.test.ts#L1589